src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1975,26 +1972,26 @@ TEST(SpatialPoolerTest, testSerialization_ar) {
     SDR outputBaseline(output);
     sp1.compute(input, true, outputBaseline);
 
-    // C - Next do old version
+    // C - Next, verify the same results come from the de/serialized version
     {
       SpatialPooler spTemp;
       testTimer.start();
 
       // Deserialize
       ss.seekg(0);
-      spTemp.load(ss);
+      EXPECT_NO_THROW(spTemp.load(ss));
 
       // Feed new record through
       SDR outputC({numColumns});
       spTemp.compute(input, true, outputC);
 
       // Serialize
       ss.clear();
-      spTemp.save(ss);
+      EXPECT_NO_THROW(spTemp.save(ss));
 
       testTimer.stop();
 
-      EXPECT_EQ(outputBaseline, outputC);
+      EXPECT_EQ(outputBaseline, outputC); //FIXME this test randomly fails. (De/serialization of rng_ is correct?) 
```

Comment:
this is another test that is inconsistent! Sometimes it passes, sometimes crashes on different outputs. I can't see why would that be, the test is relatively simple and seems correct.  
I'm suspecting something with serizalization of Random? 
Even more confusing is why HellpSPTP fails undeterministically now?